Slim/Utils/Scanner/Remote.pm

```diff
@@ -703,6 +699,118 @@ sub parseAACHeader {
 	$cb->( $track, undef, @{$pt} );
 }
 
+sub parseMp4Header {
+	my ( $http, $dataref, $track, $args, $url ) = @_;
+
+	Slim::Formats->loadTagFormatForType('mp4');
+	my $formatClass = Slim::Formats->classForFormat('mp4');
+	my $info = $formatClass->parseStream($dataref, $args);
+	
+	if ( ref $info ne 'HASH' ) {
+		if ( !$info ) {
+			$log->error("unable to parse mp4 header");
+			$args->{cb}->( $track, undef, @{$args->{pt} || []} );
+			return 0;
+		} elsif ( $info < 0 ) {
+			return 1;
+		} else {
```

Comment:
I don't see when we could be neither hash ref nor falsy, nor smaller 0. I checked callers, but didn't find one returning so?

Reply:
When mp4 parser (Slim::Format::Movie::parseStream) reaches the 'mdat' atom before and has not reached 'moov', it means that the useful header is at the end of the file/stream. So in that case the parser returns a >0 value which means "please jump to that offset and continue from there"

Reply:
Oh, that's really specific to the mp4 parser? Ok. Missed that.

Reply:
Yes, I made it a general feature, but as of today, it's only used in mp4 as hopefully, other format have header first. But I tried to build provisions for the future :-)